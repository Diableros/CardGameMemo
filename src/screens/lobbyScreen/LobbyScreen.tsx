import Alert from '../lobbyScreen/components/alert/Alert';
import StartGameBox from '../lobbyScreen/components/startGameBox/StartGameBox';
import './lobbyScreen.scss';

const SetDifficultScreen = ({ alert }: { alert: boolean }) => {
	return (
		<div className="start-screen">
			<h1 className="start-screen__header">Выбери сложность</h1>
			<StartGameBox />
			{alert ? <Alert>Выберите сложность!</Alert> : null}
		</div>
	);
};

export default SetDifficultScreen;
