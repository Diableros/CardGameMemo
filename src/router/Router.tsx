import { ReactNode } from 'react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import LobbyScreen from 'screens/lobbyScreen/LobbyScreen';
import PlayScreen from 'screens/playScreen/PlayScreen';
import NotFoundScreen from 'screens/notFoundScreen/notFoundScreen';

const Router = ({ children }: { children: ReactNode }) => {
	const router = createBrowserRouter([
		{
			path: '/',
			element: children,
			children: [
				{
					path: 'lobby',
					element: <LobbyScreen />,
				},
				{
					path: 'play',
					element: <PlayScreen />,
				},
			],
			errorElement: <NotFoundScreen />,
		},
	]);

	return <RouterProvider router={router} />;
};

export default Router;
