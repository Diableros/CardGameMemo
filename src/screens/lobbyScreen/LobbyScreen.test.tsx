import renderer from 'react-test-renderer';
import Router from 'router/Router';
import LobbyScreen from './LobbyScreen';

it('renders correctly', () => {
	const tree = renderer
		.create(
			<Router>
				<LobbyScreen />
			</Router>
		)
		.toJSON();
	/*eslint no-undef: "off"*/ expect(tree).toMatchSnapshot();
});
