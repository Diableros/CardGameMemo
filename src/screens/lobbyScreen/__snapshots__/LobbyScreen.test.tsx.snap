// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="start-screen"
>
  <h1
    className="start-screen__header"
  >
    Выбери сложность
  </h1>
  <div
    className="start-screen__difficult-box"
  >
    <div
      className="start-screen__difficult-button hover-scale"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="start-screen__difficult-button hover-scale"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="start-screen__difficult-button hover-scale"
      onClick={[Function]}
    >
      3
    </div>
  </div>
  <button
    className="start-screen__start-button hover-scale"
    onClick={[Function]}
  >
    Старт
  </button>
  <div
    className="alert"
  >
    <h1
      className="alert__message"
    >
      Выберите сложность!
    </h1>
  </div>
</div>
`;
